ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/MaximumLagFilter.java

```diff
@@ -66,8 +66,8 @@ public boolean filter(final KsqlHostInfo hostInfo) {
           final long offsetLag = Math.max(endOffset - hostLag.getCurrentOffsetPosition(), 0);
           return offsetLag <= allowedOffsetLag;
         })
-        // If we don't have lag info, we'll be conservative and include the host
-        .orElse(true);
+        // If we don't have lag info, we'll be conservative and not include the host
```

Comment:
lol - which one's conservative: including or excluding? ;)

Reply:
:).In addition would be good to capture the intent here.

Reply:
I'll update the comments a bit.  The question as to what's more conservative has to do with whether we're favoring HA or some crude notion of consistency (enforced by lag filtering) and what additional changes might be made in light of this behavior.

In short, we were previously going to forward the lag checking to the active/standby host if the router host didn't have lag info.  In this sense, it was honoring HA to let it pass through and hope the other host could make a lag decision.  Unfortunately, after discussing, it turns out that doing this properly under various scenarios is more complex than it seems.  So, to honor the lag filtering, the simplest approach is to wait until lags exist at the router, and to maximize HA by minimizing this missing lag window.

Hopefully that summary sheds a little light on the comment changes!  :-)